lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,805 @@
+                    scanner_files = scanner_stats["total_files"]
+                else:
+                    scanner_files = 0
+                    for root_info in scanner_stats["roots"]:
+                        scanner_files += root_info["files"]
+            if "dbdump_before" in fstats:
+                dbdump_before_files = fstats["dbdump_before"]["files"]
+            if "dbdump_after" in fstats:
+                dbdump_after_files = fstats["dbdump_after"]["files"]
+
+        max_files_at_site = max(scanner_files, dbdump_before_files, dbdump_after_files)
+        if max_files_at_site == 0:
+            logger(logging.WARNING, '\n No files reported by scanner for this run.\
+             Will skip processing.')
+            
+        logger(logging.INFO, '\n scanner_files: %d \n dbdump_before_files: %d\
+          \n dbdump_after_files: %d \n max_files_at_site: %d' % (scanner_files, 
+          dbdump_before_files, dbdump_after_files, max_files_at_site))
+
+
+# Was the latest run ever attempted to be processed?
+        logger(logging.INFO,'\n Was the latest run %s attempted to be processed already? - %s'\
+         % (latest_run, was_cc_attempted(latest_run)))
+        if max_files_at_site > 0 and (was_cc_attempted(latest_run) is False or force_proceed is True):
+            logger(logging.INFO,'\n Will try to process the run')
+
+################################
+# Address the Dark files first
+################################
+
+# Is there another run, at least "minagedark" old, for this RSE?
+            oldenough_run = None
+            d = list_runs_by_age(path, rse, latest_run)
+            if len([k for k in d if d[k] > minagedark]) > 0:
+# i.e. there is another dark run with appropriate age
+                oldenough_run = [k for k in d if d[k] > minagedark][0]
+                logger(logging.INFO,'\n Found another run %d days older than the latest.\
+                  \n Will compare the dark files in the two.' % minagedark)
+                logger(logging.INFO,'\n The first  %d days older run is: %s'\
+                 % (minagedark,oldenough_run))
+
+                process_dark_files(path, rse, latest_run, maxdarkfraction, max_files_at_site, oldenough_run)
+            else:
+                logger(logging.INFO, '\n There is no other run for this RSE at least %d days older,\
+                 so cannot safely proceed with dark files deleteion.' % minagedark)
+
+
+###########################################
+# Finally, deal with the missing replicas
+###########################################
+
+            process_miss_files(path, rse, latest_run, maxmissfraction, max_files_at_site, oldenough_run)
+
+        else:
+# This run was already processed
+            logger(logging.INFO, '\n Nothing to do here')
+
+    else:
+# No scans outputs are available for this RSE
+        logger(logging.INFO, '\n No scans available for this RSE')
+
+
+
+def deckard_loop(rses, dark_min_age, dark_threshold_percent, miss_threshold_percent,
```

Comment:
not entirely sure what deckard means? (blade runner reference? :-)

Reply:
`deckard` is the name of the non-daemon version of the code that we developed in the last year or so, before deciding to turn this into a proper daemon. I decided to keep the function's name, as it reminds the readers of the code where it all comes from.

And - yes, it is a Blade runner reference (invented by Eric when many Rucio daemons had similar names ;)

Reply:
I like the idea of referencing the code origin, however you should use references that are clear to readers. Maybe give more context, if @mlassnig does not understand that with his experience in Rucio and the field, how should a normal mortal person be able to understand it then?